ha/cli/command_factory.py

```diff
@@ -18,22 +18,68 @@
+
+        # This dictionary contains the classes used by all HA CLIs
+        # In case of any modifications to the CLI,
+        # this dictionary should be updated.
+        self.cmd_dict = {
+            "cluster": {
+                "start": "ha.cli.exec.clusterExecutor.ClusterStartExecutor",
+                "stop": "ha.cli.exec.clusterExecutor.ClusterStopExecutor",
+                "restart": "ha.cli.exec.clusterExecutor.ClusterRestartExecutor",
+                "standby": "ha.cli.exec.clusterExecutor.ClusterStandbyExecutor",
+                "active": "ha.cli.exec.clusterExecutor.ClusterActiveExecutor",
+                "list": "ha.cli.exec.clusterExecutor.ClusterListExecutor",
+                "status": "ha.cli.exec.clusterExecutor.ClusterStatusExecutor",
+                "add": "ha.cli.exec.clusterExecutor.ClusterAddExecutor",
+                "-h": "ha.cli.exec.commandExecutor.ClusterCLIUsage",
+                "--help": "ha.cli.exec.commandExecutor.ClusterCLIUsage"
+            },
+            "node": {
+                "start": "ha.cli.exec.nodeExecutor.NodeStartExecutor",
+                "stop": "ha.cli.exec.nodeExecutor.NodeStopExecutor",
+                "standby": "ha.cli.exec.nodeExecutor.NodeStandbyExecutor",
+                "active": "ha.cli.exec.nodeExecutor.NodeActiveExecutor",
+                "status": "ha.cli.exec.nodeExecutor.NodeStatusExecutor",
+                "-h": "ha.cli.exec.commandExecutor.NodeCLIUsage",
+                "--help": "ha.cli.exec.commandExecutor.NodeCLIUsage"
+            },
+            "storageset": {
+                "start": "ha.cli.exec.storagesetExecutor.StoragesetStartExecutor",
+                "stop": "ha.cli.exec.storagesetExecutor.StoragesetStopExecutor",
+                "standby": "ha.cli.exec.storagesetExecutor.StoragesetStandbyExecutor",
+                "active": "ha.cli.exec.storagesetExecutor.StoragesetActiveExecutor",
+                "status": "ha.cli.exec.storagesetExecutor.StoragesetStatusExecutor",
+                "-h": "ha.cli.exec.commandExecutor.StoragesetCLIUsage",
+                "--help": "ha.cli.exec.commandExecutor.StoragesetCLIUsage"
+            },
+            "service": {
+                "start": "ha.cli.exec.serviceExecutor.ServiceStartExecutor",
+                "stop": "ha.cli.exec.serviceExecutor.ServiceStopExecutor",
+                "status": "ha.cli.exec.serviceExecutor.ServiceStatusExecutor",
+                "-h": "ha.cli.exec.commandExecutor.ServiceCLIUsage",
+                "--help": "ha.cli.exec.commandExecutor.ServiceCLIUsage"
+            },
+            "-h": "ha.cli.exec.commandExecutor.CLIUsage",
+            "--help": "ha.cli.exec.commandExecutor.CLIUsage"
+        }
+
+    def get_executor(self, module_name, operation_name):
+        """ return the appropriate class name from the dictionary"""
+
+        try:
+            executor = self.cmd_dict.get(module_name).get(operation_name)
+        except Exception:
```

Comment:
Can we log error before return?

Reply:
Done in the https://github.com/Seagate/cortx-ha/pull/349 PR